test/src/us/kbase/kbasefeaturevalues/test/KBaseFeatureValuesServerTest.java

```diff
@@ -329,70 +343,94 @@ public void testPyScikitKMeans() throws Exception {
+        ObjectData res = getWsClient().getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
                 .withName(clustObj1Name))).get(0);
+
+        //test if returned id matches id returned by ws saveobjects
+        Assert.assertEquals(getInfoEKN.get(0).getE7()  + "/" + getInfoEKN.get(0).getE1() + "/" +  getInfoEKN.get(0).getE5(), getId);
+
         FeatureClusters clSet = res.getData().asClassInstance(FeatureClusters.class);
         checkKMeansForSample(clSet);
     }
-    
-    @Test
+
     public void testCorrectMatrix() throws Exception {
         String testWsName = getWsName();
         String sourceMatrixId = "notcorrected_matrix.1";
+
+        FloatMatrix2D testmat = getSampleMatrix();
         ExpressionMatrix data = new ExpressionMatrix().withType("log-ratio").withScale("1.0")
-                .withData(getSampleMatrix());
+                .withData(testmat);
+        //introduce null value for imputation test
         data.getData().getValues().get(0).set(0, null);
         Assert.assertEquals(1, getNullCount(data.getData()));
-        getWsClient().saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
+        List<Tuple11<Long, String, String, String, Long, String, Long, String, String, Long, Map<String, String>>> infoCM = getWsClient().saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
                 new ObjectSaveData().withName(sourceMatrixId).withType("KBaseFeatureValues.ExpressionMatrix")
-                .withData(new UObject(data)))));
+                        .withData(new UObject(data)))));
         String targetMatrixId = "corrected_matrix.1";
-        impl.correctMatrix(new CorrectMatrixParams().withInputData(
+        String getId = impl.correctMatrix(new CorrectMatrixParams().withInputData(
                 testWsName + "/" + sourceMatrixId).withOutWorkspace(testWsName)
                 .withOutMatrixId(targetMatrixId).withTransformType("missing"), token, getContext());
+
         ExpressionMatrix matrix = getWsClient().getObjects(Arrays.asList(
                 new ObjectIdentity().withWorkspace(testWsName).withName(targetMatrixId)))
                 .get(0).getData().asClassInstance(ExpressionMatrix.class);
+
+        //test matrix imputation
         Assert.assertEquals(0, getNullCount(matrix.getData()));
-        Assert.assertEquals(0.325, (double)matrix.getData().getValues().get(0).get(0), 1e-10);
+        Assert.assertEquals(0.325, (double) matrix.getData().getValues().get(0).get(0), 1e-10);
+
+        //reload with returned object id
+        ExpressionMatrix res = loadfromWs(getId).asClassInstance(ExpressionMatrix.class);
+        //reset from null in data before imputation
+        data.getData().getValues().get(0).set(0, res.getData().getValues().get(0).get(0));
+        Assert.assertEquals(data.getData().toString(), matrix.getData().toString());
```

Comment:
This appears to be testing the input matrix against the result matrix from the method. Shouldn't they be different? My assumption is that the method is not a noop, but maybe I'm wrong.

Reply:
they should be different and one value is reset to make the comparison equal, see comments above

Reply:
Ok, so the only difference is the imputed value(s) and those are set equal to the output matrix in the input matrix, and then the matrices are compared to ensure the method has done nothing other than add the imputed values?

Reply:
that's right, with the exception that the actual imputed value itself is not tested -- will add.

Reply:
I assume these haven't been added yet? I don't see anything in the subsequent commits.

Reply:
Right, added now and in new PR:
https://github.com/realmarcin/FeatureValues/blob/5e82fbce8a75508aa8be1ad2250d18f2678039e3/test/src/us/kbase/kbasefeaturevalues/test/KBaseFeatureValuesServerTest.java#L398